src/components/CustomHooks/CustomHooks.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { ProcessGetRequest } from '../../utils/RestUtils'
+
+/**
+ * 
+ * @param {*} url For future implementation 
+ * @param {*} options For future implementation
+ */
+export default function useFetch(url, options) {
+    const [response, setResponse] = React.useState(null);
+    const [error, setError] = React.useState(null);
+    React.useEffect(() => {
+        const fetchData = async () => {
+            // setLoading(true);
+            try {
+                // Will open this for future implementations
+                //   const res = await fetch(url, options);
+                //   const json = await res.json();
+                let response = localStorage.getItem('movieTitles');
+                if (response === null || response === undefined || response.length === 0) {
+                    response = await ProcessGetRequest();
+                    if (response.status === 200) {
+                        localStorage.setItem('movieTitles', response)
```

Comment:
You must to use ```JSON.stringify(response)``` to convert to string movies object because localStorage can store only strinngs. By default Object.toString() function, which is invoced by default when you try to put object to localStorage, returns [object Object] - [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString). That's why you must to convert to string this object yourself